sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/agents/SearchKnowledgeAgentClientBuilder.java

```diff
@@ -0,0 +1,264 @@
+
+    private AzureKeyCredential azureKeyCredential;
+    private TokenCredential tokenCredential;
+    private SearchAudience audience;
+    private String endpoint;
+    private String agentName;
+    private String apiVersion;
+    private HttpClient httpClient;
+    private HttpPipeline httpPipeline;
+    private HttpLogOptions httpLogOptions;
+    private ClientOptions clientOptions;
+    private Configuration configuration;
+    private RetryPolicy retryPolicy;
+    private RetryOptions retryOptions;
+    private JsonSerializer jsonSerializer;
+
+    /**
+     * Creates a new builder instance.
+     */
+    public SearchKnowledgeAgentClientBuilder() {
+    }
+
+    /**
+     * Sets the service endpoint for the Azure AI Search instance.
+     *
+     * @param endpoint The URL of the Azure AI Search instance.
+     * @return The updated builder object.
+     */
+    @Override
+    public SearchKnowledgeAgentClientBuilder endpoint(String endpoint) {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the agent name for the Azure AI Search agent.
+     *
+     * @param agentName The name of the agent.
+     * @return The updated builder object.
+     */
+    public SearchKnowledgeAgentClientBuilder agentName(String agentName) {
+        this.agentName = agentName;
+        return this;
+    }
+
+    /**
+     * Sets the API version to use for requests.
+     *
+     * @param apiVersion The API version.
+     * @return The updated builder object.
+     */
+    public SearchKnowledgeAgentClientBuilder apiVersion(String apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder credential(AzureKeyCredential credential) {
+        this.azureKeyCredential = credential;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder credential(TokenCredential credential) {
+        this.tokenCredential = credential;
+        return this;
+    }
+
+    public SearchKnowledgeAgentClientBuilder audience(SearchAudience audience) {
+        this.audience = audience;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder httpLogOptions(HttpLogOptions logOptions) {
+        this.httpLogOptions = logOptions;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy, "'policy' cannot be null.");
+        this.perCallPolicies.add(policy); // For simplicity, treat as per-call; refine as needed
+        return this;
+    }
+
+    public SearchKnowledgeAgentClientBuilder serializer(JsonSerializer jsonSerializer) {
```

Comment:
I don't think this needs to be exposed in this builder as Knowledge Agent won't need to handle serialization of customer types (at least not now).